### ISSUE TITLE
[Question] about cas must use old_value

### ISSUE THREAD

**Issue by Nivras** [tikv/tikv#12314]
Right now compare_and_swap() should use old_value to make sure it will satisfied the CAS support. But my value is always bigger than 1M. It means if I want use tikv as a distributed cache service, I must save the old_value so that I can use compare_and_swap to make sure the cache is corrected.

So old_value is too big and it waste a lot of memory. Could tikv support some other cas interface, like use a version(int64) to replace the old_value, and version can be just the kv's mvcc version or  
other data structure that can be returned by Get(), and each compare_and_swap() need to provide a version, and tikv will check if the version is the current one. then the compare_and_swap will not return the value, can just return the version. In this way, it can save the network cost, and can save a lot of memory for cache the old value.

**Comment by BusyJay** [comment]
Good idea! For APIv2, we add a version to every raw key, so this is possible now. Do you want to write a proposal to tikv/rfcs?

**Comment by Nivras** [comment]
Sure, I will write this later, looking forward to use this feature!

**Comment by aep** [comment]
that never happened, did it? at least the go client has no versioned rawkv
